server/src/main/java/org/elasticsearch/cluster/ClusterState.java

```diff
@@ -283,7 +286,7 @@ public Map<String, TransportVersion> transportVersions() {
     }
 
     public TransportVersion getMinTransportVersion() {
-        return this.minTransportVersion;
+        return blocks.hasGlobalBlock(STATE_NOT_RECOVERED_BLOCK) ? TransportVersion.MINIMUM_COMPATIBLE : this.minTransportVersion;
```

Comment:
Why is the block check necessary when MIN_COMPATIBLE is already specified above in the init for minTransportVersion? Is the else even possible there (will the map always have at least one element, for the current node)?

Reply:
Good point, I moved the check to the constructor in ec694f153aa.

In practice the map will be nonempty (except in tests, but this requires a little thought here so I'd rather have a conservative default). The min in the map may go backwards while the cluster is forming tho:

https://github.com/elastic/elasticsearch/blob/77775e2fa42b841989db27ff5359ab29405afede/server/src/main/java/org/elasticsearch/cluster/coordination/NodeJoinExecutor.java#L129-L130